internal/rootcoord/meta_table.go

```diff
@@ -448,6 +446,50 @@ func (mt *MetaTable) AlterCollection(ctx context.Context, oldColl *model.Collect
 	return nil
 }
 
+func (mt *MetaTable) RenameCollection(ctx context.Context, oldName string, newName string, ts Timestamp) error {
+	mt.ddLock.RLock()
+	defer mt.ddLock.RUnlock()
+	ctx = contextutil.WithTenantID(ctx, Params.CommonCfg.ClusterName.GetValue())
+
+	log := log.Ctx(ctx).With(zap.String("oldName", oldName), zap.String("newName", newName))
+
+	//old collection should not be an alias
+	_, ok := mt.collAlias2ID[oldName]
+	if ok {
+		log.Warn("unsupported use a alias to rename collection")
+		return fmt.Errorf("unsupported use an alias to rename collection, alias:%s", oldName)
+	}
```

Comment:
Shall we check `newName` is an alias here? What shall happen if user rename a collection to an existing alias?

Reply:
Yes,  newName also is checked if it is an existing alias or collection name by GetCollectionByName